classes/cachemanager.php

```diff
@@ -0,0 +1,101 @@
+<?php
+// This file is part of Moodle - https://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <https://www.gnu.org/licenses/>.
+
+namespace mod_learningmap;
+
+/**
+ * Cache manager class for mod_learningmap
+ *
+ * @package     mod_learningmap
+ * @copyright   2021-2024, ISB Bayern
+ * @author      Stefan Hanauska <stefan.hanauska@csg-in.de>
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class cachemanager {
+    /**
+     * Reset the backlink cache for a course (includes rebuilding it) or the whole instance (if $courseid is 0).
+     *
+     * @param int $courseid The id of the course (defaults to 0).
+     * @return void
+     */
+    public static function reset_backlink_cache(int $courseid = 0): void {
+        $cache = \cache::make('mod_learningmap', 'backlinks');
+        if (empty($courseid)) {
+            $cache->purge();
```

Comment:
Please add a comment that this also wipes the indicator key "fillstate" which means the cache from this point on is invalidated.